source/session.ts

```diff
@@ -656,7 +658,7 @@ export class Session {
       );
     }
 
-    const primaryKeys = this.getPrimaryKeyAttributes(entityType);
+    const primaryKeys = this.getPrimaryKeyAttributes(entityType) ?? [];
```

Comment:
if primaryKeys returns an empty object, the resulting expression is not a valid expression. it is better to simply throw (or assert that getPrimaryKeyAttributes always returns a non-empty array)